src/tests/components/ForecastDetails.test.js

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import ForecastDetails from "../../components/ForecastDetails";
+
+describe("ForecastDetails", () => {
+  const validProps = {
+    humidity: 71,
+    temperature: {
+      min: 12,
+      max: 22,
+    },
+    wind: {
+      speed: 13,
+      direction: "s",
+    },
+  };
+
+  it("renders correctly", () => {
+    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it("displays the correct forecast details", () => {
+    const { getByText } = render(<ForecastDetails forecast={validProps} />);
+    expect(getByText(`Humidity: ${validProps.humidity}%`)).toBeInTheDocument();
```

Comment:
You should leave a clear line between the end of your setup and the start of the assertions, especially when the test is more than a few lines long:
```
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText(`Humidity: ${validProps.humidity}%`)).toBeInTheDocument();
```